internal/receiver/scriptedinputsreceiver/procpipe/config.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Copyright Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procpipe
+
+import (
+	"io"
+	"os/exec"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+	"go.uber.org/zap"
+)
+
+// NewConfig creates a new stdin input config with default values
+func NewConfig() *Config {
+	return &Config{
+		InputConfig: helper.NewInputConfig(operatorType, operatorType),
+		BaseConfig: BaseConfig{
+			Multiline: helper.NewMultilineConfig(),
+			Encoding:  helper.NewEncodingConfig(),
+		},
+	}
+}
+
+// Config is the configuration of a stdin input operator.
+type Config struct {
```

Comment:
https://github.com/signalfx/splunk-otel-collector/pull/3323/files#r1258319325

https://github.com/open-telemetry/opentelemetry-collector/blob/main/component/config.go#L41

You can add a `Validate()` method to ensure the script name is valid and any other checks that are currently happening on Build (`MaxLogSize` for example). Invalid configuration is a service failure and shouldn't be exposed in `Start()`